homeassistant/components/tuya/climate.py

```diff
@@ -38,6 +38,13 @@
     "wind": HVACMode.FAN_ONLY,
 }
 
+TUYA_EXTRA_ATTRS = [
+    "fault",
+    "roomtemp_calibrat",
+    "valve_set",
+    "valve",
+    "power_state"
```

Comment:
These shouldn't be attributes, but their own entities.

Reply:
one physical device = one device in HA.
What is listed above are device attributes, not separate devices.

Reply:
> These shouldn't be attributes, but their own entities.

I can understand this in the case of quiet mode in air conditioning - a separate switch, or night mode in the lamp. But these are the states we can change.

The `valwve` or `power_state` describe the state the device is in. They are neither switches nor sensors. They are just attributes of the device

Reply:
Unfortunately, our architecture does not agree. As stated, please don't use attributes.

Reply:
@frenck Thanks for the clarifications. Could you please validate my idea in the mantime regarding the enities:
- valve sensor (showing the valve position)
- power_state binary sensor
- fault binary sensor
- temperature calibration number entity (this is the main thing here, currently available only in Zigbee2MQTT, ability to easily recalibrate the built in temp sensor based on external in-room thermomenter)

Reply:
I'm not sure what you want me to validate on that? 🤷

Reply:
In your opinion, are these the right entity types

Reply:
🤷 I have no context in that little piece of text. Feel free to develop and test it, figure out the right things and open up a PR that motivates it. That way, we can discuss it.